skimage/exposure/_adapthist.py

```diff
@@ -244,7 +244,8 @@ def clip_histogram(hist, clip_limit):
     n_excess -= mid.size * clip_limit - mid.sum()
     hist[mid_mask] = clip_limit
 
-    while n_excess > 0:  # Redistribute remaining excess
+    iteration = 0
+    while n_excess > 0 and iteration < 20:  # Redistribute remaining excess
```

Comment:
Agreed. Also, as I mentioned in the Issue previously, I think the problem may resolve without modification of the underlying code if we simply check and ensure `clip_level` is greater than the image minimum value.